util/pace/util/_legacy_restart.py

```diff
@@ -111,13 +119,13 @@ def _apply_dims(da, new_dims):
 def _apply_restart_metadata(state, restart_properties: RestartProperties):
     new_state = {}
     for name, da in state.items():
-        if name in restart_properties.keys():
-            properties = restart_properties[name]
-            new_dims = properties["dims"]
-            new_state[name] = _apply_dims(da, new_dims)
-            new_state[name].attrs["units"] = properties["units"]
-        else:
-            new_state[name] = copy.deepcopy(da)
+        for key in restart_properties.keys():
+            if name in restart_properties[key]["driver_name"]:
```

Comment:
Why was this for-if flipped? Or, why was this function changed?

Reply:
restart_properties.keys() are things like `x_wind`, `x_wind_on_c_grid`, and would not match the names in `state.items()`